### PR TITLE
1762 nested maps with bigmap root typecheck error

### LINKED ISSUE #1762: `MapTypecheckError` on `storage()` for contract `KT1UwCd8rNCATj2mwC5UgEo4uqPRVqfjRPa6` (ghostnet)
**Description**
Retrieving storage of contract `KT1UwCd8rNCATj2mwC5UgEo4uqPRVqfjRPa6` on ghostnet fails with `MapTypecheckError`.

**Steps To Reproduce**

1. Get the contract abstraction.
  ```typescript
  const toolkit = new TezosToolkit('ghostnet-node-url');
  const contract = await toolkit.contract.at('KT1UwCd8rNCATj2mwC5UgEo4uqPRVqfjRPa6');
  ```
2. Retrieve storage of the contract.
  ```typescript
  await contract.storage();
  ```
3. It fails.
    ```
    MapTypecheckError: value not compliant with underlying michelson type
        at MichelsonMap.assertTypecheckValue (...\node_modules\@taquito\michelson-encoder\src\michelson-map.ts:110:13)    
        at MichelsonMap.set (...\node_modules\@taquito\michelson-encoder\src\michelson-map.ts:166:10)
        at ...\node_modules\@taquito\michelson-encoder\src\tokens\map.ts:50:11
        at Array.forEach (<anonymous>)
        at MapToken.Execute (...\node_modules\@taquito\michelson-encoder\src\tokens\map.ts:49:9)
        at Schema.Execute (...\node_modules\@taquito\michelson-encoder\src\schema\storage.ts:143:31)
        at RpcContractProvider.<anonymous> (...\node_modules\@taquito\taquito\src\contract\rpc-contract-provider.ts:85:27)
        at Generator.next (<anonymous>)
        at fulfilled (...\node_modules\@taquito\taquito\dist\taquito.umd.js:41:62)
    ```

### PR DESCRIPTION
**Pull request by zainen**
closes #1762

### CODE REVIEW

File: packages/taquito/src/contract/rpc-contract-provider.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class RpcContractProvider
   constructor(context: Context, private estimator: EstimationProvider) {
     super(context);
   }
-  contractProviderTypeSymbol = Symbol.for('taquito--provider-type-symbol');
+  contractProviderTypeSymbol = Symbol.for('taquito--provider-type-symbol'); // missing a word? taquito--provider...
```
Comment (roxaneletourneau) [inline packages/taquito/src/contract/rpc-contract-provider.ts:74]: taquito-contract-provider-type-symbol maybe?

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 2]
```diff
@@ -2203,4 +2203,20 @@ describe('RpcContractProvider test', () => {
       done();
     });
   });
-});
+  describe('map typecheck error', () => {
+
+    it('should have defined storage with TicketTokens without errors (#1762)', async (done) => {
```
Comment (roxaneletourneau) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:2208]: I think `TicketTokens` in the title is wrong. Is it not to test Bigmap inside map?

Comment (zainen) [reply packages/taquito/test/contract/rpc-contract-provider.spec.ts:2208]: right! forgot to change after copy paste

File: packages/taquito-michelson-encoder/src/schema/storage.ts [hunk 3]
```diff
@@ -165,6 +166,9 @@ export class Schema {
     if (this.root instanceof TicketDeprecatedToken && val.ticketer && val.value && val.amount) {
       return true;
     }
+    if (this.root instanceof MapToken && this.root.ValueSchema instanceof BigMapToken) {
```
Comment (dsawali) [inline packages/taquito-michelson-encoder/src/schema/storage.ts:169]: I understand the MapToken check, but why do we need to also check `this.root.ValueSchema`?

File: packages/taquito/test/helpers.ts [hunk 4]
```diff
@@ -591,3 +591,89 @@ export const ticketTokenTestMock = {
     ],
   },
 };
+
+export const smallerTypeCheckError = {
```
Comment (dsawali) [inline packages/taquito/test/helpers.ts:595]: might want to rename this to something a bit more descriptive, it isn't clear what a smaller typecheck error is

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 5]
```diff
@@ -35,7 +35,7 @@ import { OpKind, ParamsWithKind, TransferTicketParams } from '../../src/operatio
 import { NoopParser } from '../../src/taquito';
 import { OperationBatch } from '../../src/batch/rpc-batch-provider';
 import { ContractMethodObject } from '../../src/contract/contract-methods/contract-method-object-param';
-import { ticketTokenTestMock } from '../helpers';
+import { anotherMapTypecheckError, smallerTypeCheckError, ticketTokenTestMock } from '../helpers';
```
Comment (dsawali) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:38]: same here with the naming, `anotherMapTypecheckError` doesn't sound very descriptive

**Review by dsawali — CHANGES_REQUESTED** [review]

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts (hunk 5 above)
Comment (dsawali) [reply packages/taquito/test/contract/rpc-contract-provider.spec.ts:38]: it's also not used anywhere it seems, it's failing the linter

File: packages/taquito-michelson-encoder/src/schema/storage.ts (hunk 3 above)
Comment (zainen) [reply packages/taquito-michelson-encoder/src/schema/storage.ts:169]: so far map typecheck as been fine its just when checking the type of a BigMapToken

Comment (zainen) [reply packages/taquito-michelson-encoder/src/schema/storage.ts:169]: nested inside

**Comment by dsawali** [comment]
could you add a closes # ticket number to link this PR to the ticket

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**